distribution/debian/scripts/postinst

```diff
@@ -1,6 +1,8 @@
 #!/bin/sh
 set -e
 
+usermod -aG systemd-journal vector || true
```

Comment:
I prefer the long form of flags in scripts to make it less likely I have to go lookup what a flag does. In this case, that'd be: `usermod --append --groups systemd-journal vector`.

I take it this group doesn't always exist on these systems given the `|| true`? Do we need different permissions for `vector` to be able to use the `journald` source on those OSes?

Reply:
This isn't user-facing and a short form is consistently used elsewhere in our packaging so I am going to keep it. 

I added the || true in the unlikely event someone tried to install the packages on a platform without systemd (people do force installs etc on older platforms for example).

Reply:
> This isn't user-facing and a short form is consistently used elsewhere in our packaging so I am going to keep it.

I think we should be preferring long form flags elsewhere in this script (and generally in checked-in scripts). I realize this could be bike shedding a little, but I think there is a concrete practical benefit to preferring long-form flags when they exist: namely that it is less likely that the reader of a script will need to lookup what a short-flag does if they are unfamiliar with it. In this particular example, `--append` and `--groups` are clear enough that, even if I wasn't familiar with `usermod`, I'd probably be able to guess what it does.

I think the short-form flags are useful when used in an interactive terminal to save keystrokes, but for checked-in scripts that are likely to be read often, I think we should prefer legibility over typing speed.

I'd propose that all new shell script code use the long form flags, when they exist, and that we switch to them as we modify existing code.

> I added the || true in the unlikely event someone tried to install the packages on a platform without systemd (people do force installs etc on older platforms for example).

:+1: